### ISSUE TITLE
[3.8.x] CreateSequence fails on Postgres <9.4 due to IF NOT PRESENT

### ISSUE THREAD

**Issue by jpmaas** [liquibase/liquibase#1039]
**Description**
Due to recent changes in liquibase 3.8.x, the statement to create a sequence is as follows:
`CREATE SEQUENCE IF NOT EXISTS <schema>.<sequenceName> <further configuration>`.

This is fine for versions >= 9.5. In versions below that, the IF NOT EXISTS syntax is not supported.

**To Reproduce**
Add changelog to create a sequence and execute it.

```
<changeSet author="jpmaas" id="identifier">
        <preConditions onFail="MARK_RAN">
            <not>
                <sequenceExists sequenceName="sequence"/>
            </not>
        </preConditions>
        <createSequence sequenceName="sequence" startValue="1"/>
    </changeSet>
```

In my case, this changeset is executed via application startup using spring boot 2.

During startup, the following exception is thrown (starting from cause because spring exception is not necessary):

```
Caused by: liquibase.exception.MigrationFailedException: Migration failed for change set db/changelog.xml::identifier::jpmaas:
     Reason: liquibase.exception.DatabaseException: ERROR: syntax error at or near "NOT"
  Position: 21 [Failed SQL: (0) CREATE SEQUENCE  IF NOT EXISTS schema.sequence START WITH 1]
        at liquibase.changelog.ChangeSet.execute(ChangeSet.java:646)
        at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:53)
        at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:83)
        at liquibase.Liquibase.update(Liquibase.java:202)
        at liquibase.Liquibase.update(Liquibase.java:179)
        at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:366)
        at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:314)
        at liquibase.integration.spring.MultiTenantSpringLiquibase.runOnAllSchemas(MultiTenantSpringLiquibase.java:147)
        at liquibase.integration.spring.MultiTenantSpringLiquibase.afterPropertiesSet(MultiTenantSpringLiquibase.java:94)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1855)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1792)
        ... 24 common frames omitted
Caused by: liquibase.exception.DatabaseException: ERROR: syntax error at or near "NOT"
  Position: 21 [Failed SQL: (0) CREATE SEQUENCE  IF NOT EXISTS schema.sequence START WITH 1]
        at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:402)
        at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:59)
        at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:131)
        at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1274)
        at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1256)
        at liquibase.changelog.ChangeSet.execute(ChangeSet.java:609)
        ... 34 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: syntax error at or near "NOT"
  Position: 21
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2497)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2233)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:310)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:446)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:370)
        at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:311)
        at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:297)
        at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:274)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:269)
        at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:95)
        at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java)
        at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:398)
        ... 39 common frames omitted

```

**Expected behavior**
The sequence can be created without exception on postgres versions below 9.5.

**Additional context**
The code in question is located at this position:
https://github.com/liquibase/liquibase/blob/master/liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateSequenceGenerator.java#L50

┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-150) by [Unito](https://www.unito.io/learn-more)

**Comment by SteveDonie** [comment]
Thanks for the bug report! 

I'm not a Postgres expert - what is the correct SQL for Postgres < 9.5? Is it just `CREATE SEQUENCE schema.sequence START WITH 1`?

Is it important to you that this be fixed in the 3.8.x branch, or could this be included in the currently-evolving master branch which will soon be 4.0.x?

**Comment by jpmaas** [comment]
Yes, it is just `CREATE SEQUENCE schema.sequence START WITH 1`. This is also the result of older liquibase versions (<=3.7.0) which works perfectly fine.

Regarding the fix version: I think it might be useful in 3.8.x since it is used in other major frameworks like spring. But if this is the only fix for that branch, it feels like too much overhead. But since I don't know your timeline for 4.0.x, you should decide what's best for a fix I guess.

**Comment by nfalco79** [comment]
We have the same issue. With liquibase 3.5.5 work. Moving on 3.8.9 no more. As PostgreSQL documents the IF NOT EXISTS statement is available since 9.5
* https://www.postgresql.org/docs/9.5/sql-createsequence.html
* https://www.postgresql.org/docs/9.4/sql-createsequence.html
